backends/build_system/validate_env.py

```diff
@@ -30,12 +31,16 @@
 
 class UnmetDependenciesException(Exception):
     def __init__(self, unmet_deps):
+        pip_install_command = f"{sys.executable} -m pip install "
         msg = "Environment requires following Python dependencies:\n\n"
         for package, actual_version, required in unmet_deps:
             msg += (
                 f"{package} (required: {required.constraints}) "
                 f"(version installed: {actual_version})\n"
             )
+            pip_install_command += f'"{package}{required.string_constraints()}" '
+        msg += "\nRun the following pip command to install manually:\n"
```

Comment:
Looking at the output from the overview from the PR, I'm a little concerned that the error message is confusing in terms of what to do if someone specifically gets that error. Mainly this error message says to install the dependencies manually but we also have the other error message that says to consider using the `--with-download-deps` flag. So I can see someone being confused on which approach to use. In general, it would be great if we can expand this error message to:

1. Recommend `--with-download-deps` as the top option since that is likely the quickest way to get this working and it won't impact any system installed dependencies
2. Recommend the pip command if `--with-download-deps` does not work or they actually want to use the currently installed Python package
3. Bonus idea - The error detects if the Python interpreter is part of a virtaulenv and gives a warning for the pip commands if the global interpreter is being used and potentially recommends to do all of this setup from within a virtualenv.

Thoughts?